src/dune_rules/dialect.ml

```diff
@@ -227,21 +251,26 @@ module DB = struct
   type t =
     { by_name : dialect String.Map.t
     ; by_extension : dialect String.Map.t
-    ; mutable extensions_for_merlin : string option Ml_kind.Dict.t list option
+    ; for_merlin : for_merlin Lazy.t
+    }
+
+  and for_merlin =
+    { extensions : string option Ml_kind.Dict.t list
+    ; extension_to_reader : (string * string list) list
```

Comment:
If I understand correctly, this is supposed to map file extensions to lists of "readers". Wouldn't it be better to use a Map here? Also, I find the name a bit unintuitive; what about simply `readers`?

Reply:
Can use map but in practice the number of dialects is negligible I think...

Reply:
I wasn't suggesting it because of the number of dialects, rather because it more clearly exposes the "map" semantics of the field (eg the keys are unique).

Reply:
I see, that's what the field name was supposedly to clarify. Anyway, I've changed it as suggested (and the version check as well).